libextract/coretools.py

```diff
@@ -20,7 +20,12 @@ def proc(pairs):
     return proc
 
 
-def pipeline(data, functions):
-    for item in functions:
-        data = item(data)
-    return data
+def pipeline(data, *args):
+    try:
+        for arg in args[0:]:
+            data = arg(data)
+    except(TypeError):
```

Comment:
**Update:** I do understand why we need this, but its still less readable compared to passing a data structure in:

```python
pipeline(fp, [func1, func2])
# compared to
pipeline(fp, func1, func2)
```

And avoids unnecessary magic. Arguably more efficient as it avoids catching exceptions.